### ISSUE TITLE
Add a way for controllers to handle async exceptions

### ISSUE THREAD

**Issue by Mick605** [oas-tools/oas-tools#344]
### TL;DR
When using async endpoints inside controllers with Express 4, thrown errors cannot be handled because of the lack of the `next` argument

### Description
When using synchronous controllers, the error is catched by the error middleware just fine
```js
export function syncEndpointThrow(req, res) {
    throw new Error(`Error`)
}
```
```
[oas-tools] ERROR: Error: Error
```

When doing the same thing with an async function, the error is not handled the same way and results in a unhandled error
```js
export async function asyncEndpointThrow(req, res) {
    throw new Error(`Error`)
}
```
```
UnhandledPromiseRejectionWarning: Error: Error
```
**This is normal, because by default Express 4 does not handle async exception.**
The code should be instead

```js
export async function asyncEndpointThrow(req, res, next) {
    try {
        throw new Error(`Error`)
    } catch (err) {
        next(err) // but next is undefined
    }
}
```

The problem is that the router middleware from `oas-tools` does not provide the `next` argument to the controller, so we have no solution to handle async promises.

One solution could be to pass the next argument to the controller
```js
  // file middleware/oas-router.js, row 16
  static async initialize(oasFile, config) {
    const controllers = await OASRouter.#loadControllers(oasFile, config);
    return new OASRouter(oasFile, (req, res, _next) => {  
      const requestPath = req.route.path;
      const method = req.method;
      controllers[requestPath][method](req, res); // <-- shoud be controllers[requestPath][method](req, res, _next); 
    });
  }
```
Another (better) solution could be to handle the async error and call next automatically inside the router (Express 5 do it that way)

Is it possible to implement one of these solutions ?

Thank you in advance

**Comment by alexmathew98** [comment]
Any update on when this would be released? I would need this for a project im working on.
